### ISSUE TITLE
Castle screen, Warlock moat is missing a part behind Captain's quarters.

### ISSUE THREAD

**Issue by Branikolog** [ihhub/fheroes2#4658]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

The part of moat is not visible in the area, Captain's quarters is supposed to appear.
![Warl](https://user-images.githubusercontent.com/55348946/142408249-4e62ac4d-9bdb-4587-938b-dfe898a5f57d.png)

### Save file

Run any game and build moat.

**Comment by zenseii** [comment]
Is picture above what it looks like in both the original game and fheroes2?

**Comment by Branikolog** [comment]
Hi, @zenseii !
Yes. Both games have this flaw.

**Comment by Branikolog** [comment]
Hi, @1GodRage !

Nice try! Actually, I haven't noticed a hill near castle before. So, right now I cannot say moat is cut and the missing part should be added. But here's anyway a bad design of that part. Probably, if the hill itself could be done more perceptible, we don't need to draw missing part of the moat, since it's hidden behind the hill.
![Hill](https://user-images.githubusercontent.com/55348946/190594077-43c2cad6-ce94-4c20-a542-09af09421c6a.png)
Could it be better to make hillside more evident by adding darker/brighter pixels there? Also, a place, where the moat is cut should be more crisp. How do you think?

**Comment by zenseii** [comment]
> The water is hidden by the hill... But, if we remove the hill... ![fix2](https://user-images.githubusercontent.com/63362446/190480614-d803d359-0905-4487-8b0d-a1177c7b075e.png) or maybe you want it perfectly circular around the castle? and then remove the hill? this shape maybe: ![fix3](https://user-images.githubusercontent.com/63362446/190482345-1b04cc2e-962d-42d1-85c5-619c57ff11af.png)

Nevertheless, I just want to say I'm really impressed @1GodRage 👍

**Comment by zenseii** [comment]
> @Branikolog I think that it would be "easier" to darken the far grass, then the hill will be more visible. :) But, when the "captain" is built, there is no hill, so, I think that "deleting" the hill by making the moat in a circle shape is fine and may even give a bit of fresh air to this castle. xD I just need the asset to work with. _Me reworking on your screenshot is probably not the correct way to do it. xD_

For the assets we can't share them due to copyright. However, you can extract them from your own copy of Heroes 2 with some tools found in this repo, but you'll have to compile them and I'm not sure how familiar you are with that?

**Comment by zenseii** [comment]
Yes, basically only the different sprites between them will be needed. If you've already extracted the assets then you can probably use those.

If you still want to extract with the fheroes2 tools then you'll need to have a way to compile them and know how to do that. It isn't difficult and I could probably walk you through it. You'll also need to be somewhat familiar with navigating the console for running the extractor and converter.

Basically:
1. If you're on Windows just get something like visual studio 2017 or 2019 and install the C++ necessary stuff.
2. Download the fheroes2 source from here.
3. Setup the local repository following this guide https://github.com/ihhub/fheroes2/blob/master/docs/DEVELOPMENT.md
4. Navigate to the src/tools folder and open either `extractor-vs2017.vcxproj` or `extractor-vs2019.vcxproj`.
5. Build the debug-sdl2 version of it with visual studio and if everything has been setup right it will go smoothly.
6. Locate `extractor.exe` in the `build\x64\buildYouMade` directory. The build you made could be Debug-SDL2. I guess it could be x86.
7. Do the same again from step 4 just for `icn2img-vs2019.vcxproj`.
8. Put extractor.exe and some of the dlls found in the same build directory as it and put them all in the same folder as `HEROES2.AGG`. The dlls are `SDL2.dll`, `SDL2_mixer.dll`, `SDL2_image.dll`,  `libpng-16.dll` and `zlib1.dll`. Might be possible to build SDL1 too. You'll get a message telling you what dlls you're missing in any case.
9. Run the extractor via the console window like this `extractor path_heroes2.agg extract_to_dir` replacing the parts after extractor with the relevant paths. Example if you have heroes2.agg in the same folder as the extractor and you want to extract to a folder which will be named `extracted` in the same directory: `extractor heroes2.agg .\extracted`.
10. After successfully extracting them, you can put `icn2img.exe` in a folder with the specific file you want to extract. Make sure to have the same dlls as before present in the same folder. Then navigate to `icn2img.exe` with the console and use this structure to extract the file: `icn2img inputFile.icn destinationDirectory paletteFileName.pal`. The palette file will be inside the extracted folder and needs to be present. The filename ends with `.pal`. For example I could write` icn2img apanel.icn . kb.pal` and this will create a folder named `apanel` with the relevant images inside. Not sure exactly what icn you should look at but probably `townbkg3.icn` and `cstlwrlk.icn` ?

**Comment by Branikolog** [comment]
Hi, @1GodRage 

> @Branikolog I think that it would be "easier" to darken the far grass, then the hill will be more visible. :) But, when the "captain" is built, there is no hill, so, I think that "deleting" the hill by making the moat in a circle shape is fine and may even give a bit of fresh air to this castle. xD I just need the asset to work with. _Me reworking on your screenshot is probably not the correct way to do it. xD_

It's up to you. I personally prefer to add as little new art, as we can here. So, adding a few darker/other-color pixel over the existing hill wouldn't change the entire picture much.

**Comment by zenseii** [comment]
> it is better than before, but is it enough? or not? it is not really fancy, seems a bit clunky. I am willing to work more on it in next days.

It looks very good and a definite move in the right direction! I still feel like you need to want to see the hill to notice it. Still, looking at this gives me a much better idea of what that area of the moat is supposed to look like, compared to before.

**Comment by Branikolog** [comment]
Hi, @1GodRage 

> I still feel like you need to want to see the hill to notice it.

I agree with @zenseii , but it's really much better, than the original art, where I barely could notice a hill.

**Comment by zenseii** [comment]
> is it good? I did too much? not enough? :p
> 
> the joining part or moat/hill is quite complex, because in the moat ICN there is pixels over the hill.

I realized I forgot to reply, but it does look better. My main problem is that I've seen the castle before so to me it's sort of imprinted in my mind that there is no hill. Also I'm thinking maybe shadows could help with making it seem more like a hill? I see tht you've added some darker coloring but maybe more would make it more obvious? Similar to how you see on the other side of the castle except less due to the direction. Not sure I'm asking too much, these are just my thoughts.

**Comment by Branikolog** [comment]
Hi, @1GodRage !

I agree with. @zenseii . You can try to make it more noticeable. We can pick the best parts from both implementations to create the ultimate hill. Or roll changes back.

Talking about the last try. I think the shape is nice. But the hill looks a bit homogeneous for me. Too much gray colour spread over the entire surface of the hill. To make it more noticeable, I think you should remember, how the light is drawn on a sphere: Brighter center of a visible surface, darker on the edges. That would probably add some volume.

Anyway, I like your passion and the approach of creating new content. :)

**Comment by zenseii** [comment]
> Maybe I edited too much, instead of editing, seems that next time better to post new reply. 🤔 (so, reverse mind from forum era 😀 )

Hi @1GodRage!

Sorry for not getting back to you earlier, I must've gotten too focused on the language stuff (French translation PR incoming :P). 

Anyways while looking closer now at the pictures you posted I realized that there is actually what looks like a ledge or black line on the proximal lower side of the hill.
I've tried to draw a line where it seems to go:
![image](https://user-images.githubusercontent.com/12501091/193459556-54f1779d-9aa1-45ab-ae0f-05653db61692.png)

Could it be an idea to make it even clearer?

Also I'm trying to understand if that hill is concave or convex on the side that we can see. By concave I mean something like a sand dune. This could make sense in how they might've flattened the terrain in order to build the castle there.

Here's what I mean by concave:
![image](https://user-images.githubusercontent.com/12501091/193459914-f8f39d7a-53c9-41dd-a8af-8a54fed41246.png)


And concave:
![image](https://user-images.githubusercontent.com/12501091/193459945-13cfc214-de88-4ab7-b628-6fb165ee66ff.png)


As for the green color chosen I think it's a bit too bright. If compared to other areas the only bright part of the picture is the one to the left of the tent:
![image](https://user-images.githubusercontent.com/12501091/193459730-ee3f37f5-52cd-49be-ad13-7f21a3dbc096.png)


To me this almost gives the impression that there is a tall mountain off to the right side that doesn't cast it's shadow in that area behind to the left of the tent but covers the rest of the castle ground. There are indeed tall mountains in the background but also the centaurs' dwelling suggests this. 

However, it might also just be the weather since we can see that the sky is very cloudy, but with some open spots that might allow for some sunlight to come through, and this might be happening behind the tent...

I think in the end a lot is up to the imagination here.

**Comment by zenseii** [comment]
> So... is it concave of convex? Coud it be concave on the left-half and convex on the right-half?...
> 
> _I think it would be fancier to just remove the hill, like what I did in my first reply in this thread._

Yes I agree. I'd prefer the one where it is flattened and removed. In other words this:
![image](https://user-images.githubusercontent.com/12501091/204256758-31089616-0d24-4a05-883e-ffd09766c7f0.png)
:

**Comment by minuspinus** [comment]
> ok ^^ It would be good to save the guide you wrote [@zenseii](https://github.com/zenseii) , maybe in the wiki of this repo? ^^ Cause I'm not going to do that today, but, maybe later. I will use the "easy to use" ironfist-tools to extract it, (it should give the same resulting images.)
> 
> > Basically:
> 
> __very complex method__
> 
> Here is where I'm right now. ![actualsprite](https://user-images.githubusercontent.com/63362446/190925040-b69ef68e-8960-4773-ac7a-b20f7cccdb93.png) ![town+somefixes-nomoat](https://user-images.githubusercontent.com/63362446/190924984-b46ee052-7adc-4492-a94b-3d766e215d14.png) ![town+somefixes](https://user-images.githubusercontent.com/63362446/190924982-5b6ee0ee-3908-4ffd-a33f-bc8412288cc1.png) ![castle+somefixes](https://user-images.githubusercontent.com/63362446/190924986-ad428aa7-2c35-4d9f-8cc4-12d0af5fee45.png)
> 
> it is better than before, but is it enough? or not? it is not really fancy, seems a bit clunky. I am willing to work more on it in next days.

The tent shadow is wrong here, it's in the other direction. 

![Image](https://github.com/user-attachments/assets/9d2a6ba2-9e8c-4b45-aa3a-4e79d5d7fb27)

**Comment by minuspinus** [comment]
> [@minuspinus](https://github.com/minuspinus), please open a separate issue for this because it doesn't concern the topic of this issue.

https://github.com/ihhub/fheroes2/issues/9683
